korok/rerankers/rerankers.py

```diff
@@ -1,18 +1,30 @@
+import importlib
 from typing import List, Sequence, Tuple
-
-from sentence_transformers import CrossEncoder
-
+    
 
 class CrossEncoderReranker:
+    """A reranker class that uses a cross-encoder to rerank the results."""
+
     def __init__(self, model_name: str = "cross-encoder/ms-marco-MiniLM-L-6-v2", max_length: int = 512) -> None:
         """
         Initialize the CrossEncoder reranker.
 
         :param model_name: The name of the pre-trained cross-encoder model.
         :param max_length: The maximum sequence length for the cross-encoder.
         """
+        if not self.is_available():
```

Comment:
IMO, checking whether an import is available is not a responsibility of the class. If you would like to tie requirements directly to classes, I would add a list of requirements to the class, which can then be imported/checked by a utility function.